packages/qri-node/lib/qri-node.js

```diff
@@ -1,25 +1,41 @@
 const fs = require('fs-extra')
-const { spawn } = require('child_process');
+const { spawnSync } = require('child_process');
 
 module.exports = {
+  /**
+  * 'initialize' creates a new dataset, links it to the current directory,
+  * and creates starter files for the dataset's components
+  * @param {string} dirPath - The path to create the new dataset in
+  */
   init: (dirPath) => {
     console.log(`initializing Qri Dataset in ${dirPath}`)
     // make an empty dir
     fs.emptyDirSync(dirPath)
 
     return new Promise((resolve, reject) => {
       try {
-        const child = spawn(`cd ${dirPath} && qri init`, [], { shell: true, stdio: [process.stdin, process.stdout, process.stderr] })
-        child.on('exit', () => {
-          resolve()
-        })
+        const { status, stdout, stderr } = spawnSync(`cd ${dirPath} && qri init`, [], { shell: true })
+
+        if (status === 1) {
```

Comment:
Instead of checking `status === 1` for failure, check if `status === 0` for success. Command shells defined 0 to mean success and anything else to be an error, 1 is not guaranteed to be used.

Same for all of the spawnSync calls in this file.

Reply:
👍 I will also centralize the spawnsync so all of the qri methods are just passing commands to a central `executeQriCommand()` or whatever